### PR TITLE
Add net.sourceforge.lgames.LBreakoutHD game

### PR DESCRIPTION
**Pull request by nbenitez**
### Please confirm your submission meets all the criteria

- [X] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [X] My pull request follows the instructions at [App Submission][submission].
- [X] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [X] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [X] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:** https://sourceforge.net/p/lgames/support-requests/3/
- [X] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [X] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

File: net.sourceforge.lgames.LBreakoutHD.yaml [hunk 1]
```diff
@@ -0,0 +1,46 @@
+app-id: net.sourceforge.lgames.LBreakoutHD
+runtime: org.freedesktop.Platform
+runtime-version: '21.08'
+sdk: org.freedesktop.Sdk
+finish-args:
+  - --device=dri
+  - --share=ipc
+  - --socket=x11
+  - --socket=pulseaudio
+  - --filesystem=/app/var/lbreakouthd.hscr
```
Comment (hfiguiere) [inline net.sourceforge.lgames.LBreakoutHD.yaml:10]:
```suggestion
```

`/app` is always accessible. Also read only.

File: net.sourceforge.lgames.LBreakoutHD.yaml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  - '*.a'
+modules:
+  - shared-modules/glu/glu-9.json
+  - shared-modules/SDL/SDL-1.2.15.json
+  - shared-modules/SDL/SDL_mixer-1.2.12.json
+  - shared-modules/SDL/SDL_image-1.2.12.json
+  - shared-modules/SDL/SDL_ttf-2.0.11.json
+  - name: lbreakouthd
+    buildsystem: simple
+    build-commands:
```
Comment (hfiguiere) [inline net.sourceforge.lgames.LBreakoutHD.yaml:29]: why not using `buildsystem: autotools`.

File: net.sourceforge.lgames.LBreakoutHD.desktop [hunk 3]
```diff
@@ -0,0 +1,11 @@
+[Desktop Entry]
```
Comment (hfiguiere) [inline net.sourceforge.lgames.LBreakoutHD.desktop:1]: upstream has a desktop file, so why adding one here?

https://sourceforge.net/p/lgames/code/HEAD/tree/trunk/lbreakouthd/lbreakouthd.desktop.in

**Review by hfiguiere — CHANGES_REQUESTED** [review]
Also, you should submit patches upstream for the appdata and the icon in a proper format.

**Comment by hfiguiere** [comment]
bot, build net.sourceforge.lgames.LBreakoutHD

**Comment by hfiguiere** [comment]
Build failed on aarch64

You need to add a "source" like this to copy over the distro config.sub and config.guess to replace the old one from upstream:
https://github.com/flathub/org.gimp.GIMP/blob/74242e5908504cc9b66434f20b8881662ec13c5e/org.gimp.GIMP.json#L254

Combined with the use of "autotools" as buildsystem as commented.

File: net.sourceforge.lgames.LBreakoutHD.desktop (hunk 3 above)
Comment (nbenitez) [reply net.sourceforge.lgames.LBreakoutHD.desktop:1]: That one is very basic, lacking keywords and categories that would make the game more easy to find on flathub.org

**Comment by nbenitez** [comment]
> Build failed on aarch64
> 
> You need to add a "source" like this to copy over the distro config.sub and config.guess to replace the old one from upstream: https://github.com/flathub/org.gimp.GIMP/blob/74242e5908504cc9b66434f20b8881662ec13c5e/org.gimp.GIMP.json#L254
> 
> Combined with the use of "autotools" as buildsystem as commented.

Thanks, better than manually download those files as I was in the midst of doing.

**Comment by nbenitez** [comment]
bot, build net.sourceforge.lgames.LBreakoutHD

File: net.sourceforge.lgames.LBreakoutHD.desktop (hunk 3 above)
Comment (hfiguiere) [reply net.sourceforge.lgames.LBreakoutHD.desktop:1]: 1. it doesn't changed the search on flathub. only the appdata matters
2. `desktop-file-edit` can be used in the post-install to add things to upstream .desktop file as needed.

**Comment by hfiguiere** [comment]
bot, build net.sourceforge.lgames.LBreakoutHD

**Comment by hfiguiere** [comment]
urgh, sorry I missed it was already requested.

File: net.sourceforge.lgames.LBreakoutHD.desktop (hunk 3 above)
Comment (nbenitez) [reply net.sourceforge.lgames.LBreakoutHD.desktop:1]: > 1. it doesn't changed the search on flathub. only the appdata matters

I think you're wrong, the keywords defined in the .desktop file are connected to the search on flathub, i.e. you can search any app on flathub.org by the keywords in the app .desktop file.

Comment (hfiguiere) [reply net.sourceforge.lgames.LBreakoutHD.desktop:1]: even if I am, 2. still stands. And then you can submit a patch upstream.

Comment (nbenitez) [reply net.sourceforge.lgames.LBreakoutHD.desktop:1]: Submitted -> https://sourceforge.net/p/lgames/patches/29/

**Comment by nbenitez** [comment]
bot, build net.sourceforge.lgames.LBreakoutHD

**Comment by pizzadude** [comment]
> Build [84582 successful](https://flathub.org/builds/#/builders/32/builds/84582) To test this build, install it from the testing repository:
> 
> ```
> flatpak install --user https://dl.flathub.org/build-repo/82441/net.sourceforge.lgames.LBreakoutHD.flatpakref
> ```

Works for me! It should be worth noting that themes and levelsets can be inserted into the folder ~/.var/app/net.sourceforge.lgames.LBreakoutHD/.lbreakouthd/ . Maybe mention this in the flatpak description, with a link to the original site where you get the themes / levelsets?

**Comment by nbenitez** [comment]
> can be inserted into the folder ~/.var/app/net.sourceforge.lgames.LBreakoutHD/.lbreakouthd/

You can just use `~/.lbreakouthd/` which is also valid and more expected location.

**Comment by pizzadude** [comment]
> > can be inserted into the folder ~/.var/app/net.sourceforge.lgames.LBreakoutHD/.lbreakouthd/
> 
> You can just use `~/.lbreakouthd/` which is also valid and more expected location.

The flatpak as it is right now does not recognize the files in ~/.lbreakouthd

**Comment by hfiguiere** [comment]
that's because it does `--persist=.lbreakouthd`

**Pull request closed without merging**